From 18b7222f47c052e3ecd11ff6004a9ee6c30e0dc5 Mon Sep 17 00:00:00 2001
Message-Id: <18b7222f47c052e3ecd11ff6004a9ee6c30e0dc5.1429601425.git.chang-joon.lee@intel.com>
In-Reply-To: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
References: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Thu, 16 Apr 2015 21:55:04 +0530
Subject: [PATCH 7/9] SQUASHME! [VPG]: drm/i915: Update default wm in 3 planes
 mode.

This patch updates the default watermarks in case of 3planes
enabled. This is a power optimization as the calculated values
are less optimal.

SQUASHME: This patch needs to me squashed with
	commit e8c876f7944d03b2748c9afe355ba57d6d313e25
	Author: vandita kulkarni <vandita.kulkarni@intel.com>
	Date:   Fri Apr 10 21:10:24 2015 +0530

		REVERTME: [VPG]: drm/i915 : program dynamic dsparb and wm

Issue: GMINL-4468
Change-Id: I79cdfb7a00f88eca54874b236353963be03d184d
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index bce9050..a5dfc83 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1633,11 +1633,14 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 		fifo_size = (32 * 1024 * 50) / 100;
 		pa = vlv_update_wm_val(fifo_size,
 					intel_crtc->vlv_wm.pa);
+		pa = 0x0f;
 		fifo_size = (32 * 1024 * 25) / 100;
 		sa = vlv_update_wm_val(fifo_size,
 					intel_crtc->vlv_wm.sa);
+		sa = 0x04;
 		sb = vlv_update_wm_val(fifo_size,
 					intel_crtc->vlv_wm.sb);
+		sb = 0x04;
 
 	} else if (hweight32(plane_stat) == 2) {
 		/* 2 planes, enable fifo allocation 50:50 */
-- 
1.7.9.5

